torch/_refs/__init__.py

```diff
@@ -467,21 +476,38 @@ def _isfinite(a: TensorLikeType) -> TensorLikeType:
 
 
 def _isinf(a: TensorLikeType) -> TensorLikeType:
-    # TODO Add complex tensor support to remove is_infinite prim
-    # if utils.is_complex_dtype(a):
-    #     return bitwise_or(_isinf(real(a), _isinf(imag(a))
-    # else:
-    #     return bitwise_not(bitwise_or(isnan(a), isfinite(a)))
-    if utils.is_float_dtype(a.dtype) or utils.is_complex_dtype(a.dtype):
-        return prims.is_infinite(a)
-
-    return zeros_like(a, dtype=torch.bool)
+    if utils.is_complex_dtype(a.dtype):
+        return logical_or(_isinf(real(a)), _isinf(imag(a)))
+    # TODO: use logical_not instead of bitwise_not
+    return logical_and(bitwise_not(isfinite(a)), bitwise_not(_isnan(a)))
 
 
 isinf = _make_elementwise_unary_reference(
     _isinf,
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.ALWAYS_BOOL,
-    aten_op=torch.ops.aten.isinf,  # prim/aten name mismatch
+    aten_op=torch.ops.aten.isinf,
+)
+
+
+def _isposinf(a: TensorLikeType) -> TensorLikeType:
+    return logical_and(_isinf(a), gt(a, 0))
```

Comment:
Should this just be 

```
return eq(a, float('inf'))
```

?

Reply:
I don't think that will work, see
```python
t = torch.zeros(5, dtype=torch.int)
torch.eq(t, float('inf'))
```
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [5], in <cell line: 2>()
      1 t = torch.zeros(5, dtype=torch.int)
----> 2 torch.eq(t, float('inf'))

RuntimeError: value cannot be converted to type int without overflow
```

Reply:
That seems like a bug in `eq` (and friends)

Reply:
Also for integer tensors we should short-circuit to return full_like False and not try this comparison

Reply:
Also for complex tensors isn't the definition of complex inf that either part is complex? How does that work for isposinf? Should we error on complex values to this function, or if any part is positive inf is that sufficient?

Reply:
`isposinf` is not supported for complex inputs.

Reply:
We should probably add an explicit check, then

Reply:
check for complex and short-circuit for integers added.